tools/generate_report.rb

```diff
@@ -26,7 +26,24 @@ def report_from_args
 puts "Generating report... #{report.name}"
 
 report.queue_generate_table(:userid => USER_ID)
-report._async_generate_table(MiqTask.last.id, REPORT_PARAMS)
+def report_run
+  if defined?(ManageIQPerformance)
+    ManageIQPerformance.profile('generate_report') do
+      total_report_time = Benchmark.measure { yield }
+      puts "Total report time: #{total_report_time.utime} (user)  #{total_report_time.stime} (system)  #{total_report_time.real} (real)  #{total_report_time.total} (total)"
+      puts "Total time in SQL: #{Thread.current[:miq_perf_sql_query_data][:queries].inject(0){ |s,r| s + r[:elapsed_time] }.round(2)} ms"
+      puts "Total count queries #{Thread.current[:miq_perf_sql_query_data][:total_queries]}"
+      slowest_query = Thread.current[:miq_perf_sql_query_data][:queries].max_by{ |x| x[:elapsed_time] }
+      puts "Slowest SQL query took #{slowest_query[:elapsed_time]} ms: \n" + slowest_query[:sql]
```

Comment:
I understand "why" you had to access things via the Thread variables, but I don't like that you had to...

Again, I think if we can return a hash from the `.profile` method with all of the data that we collected, separated by type, I think this code can return more accurate timings and use common code that can be used else where.